### PR TITLE
SFTP to Google Cloud Storage Transfer system tests migration (AIP-47)

### PR DESCRIPTION
**Pull request by bkossakowska**
SFTP to Google Cloud Storage Transfer system tests migration (AIP-47)

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

File: tests/system/providers/google/cloud/gcs/example_sftp_to_gcs.py [hunk 1]
```diff
@@ -21,39 +21,60 @@
...
+LOCAL_PATH = str(Path(CURRENT_FOLDER) / "resources")
+
+FILE_LOCAL_PATH = str(Path(LOCAL_PATH) / TMP_PATH / DIR)
+FILE_NAME = "tmp.tar.gz"
 
 
 with models.DAG(
-    "example_sftp_to_gcs",
+    DAG_ID,
     start_date=datetime(2021, 1, 1),
```
Comment (bhirsz) [inline tests/system/providers/google/cloud/gcs/example_sftp_to_gcs.py:55]: Missing schedule = "@once", it won't trigger without it

File: tests/system/providers/google/cloud/gcs/example_sftp_to_gcs.py [hunk 2]
```diff
@@ -22,39 +22,61 @@
...
 
 OBJECT_SRC_1 = "parent-1.bin"
 OBJECT_SRC_2 = "parent-2.bin"
-OBJECT_SRC_3 = "parent-3.txt"
+
+CURRENT_FOLDER = Path(__file__).parent
+LOCAL_PATH = str(Path(CURRENT_FOLDER) / "resources")
+
+FILE_LOCAL_PATH = str(Path(LOCAL_PATH) / TMP_PATH / DIR)
+FILE_NAME = "tmp.tar.gz"
```
Comment (bhirsz) [inline tests/system/providers/google/cloud/gcs/example_sftp_to_gcs.py:51]: Do we need to use zipped file for testing transfering the data between SFTP and Google storage? Or maybe it could be an empty txt file (and then we can also remove BashOperator that unzip the file).

Comment (bkossakowska) [reply tests/system/providers/google/cloud/gcs/example_sftp_to_gcs.py:51]: tmp.tar.gz is needed because we need more files for testing
